### PR TITLE
Feature user soft delete

### LINKED ISSUE #1304: Option to soft delete users

### PR DESCRIPTION
**Pull request by ashu555**
Option to soft delete users
Fixes #1304 

Run below migration 
php artisan migrate --path="database/migrations/2020_09_06_112733_add_soft_deletes_to_users_table.php"

### CODE REVIEW

File: app/Http/Controllers/Order/InvoiceController.php [hunk 1]
```diff
@@ -137,7 +137,14 @@ public function getInvoices(Request $request)
...
                             $first = $this->user->where('id', $model->user_id)->value('first_name');
                             $last = $this->user->where('id', $model->user_id)->value('last_name');
-                            $id = $this->user->where('id', $model->user_id)->first()->id;
+                            $id = $this->user->where('id', $model->user_id)->first();
+                            if ($id) {
+                                $id = $id->id;
+                            } else {
+                                $id = User::onlyTrashed()->find($model->user_id)->id;
+                                $first = User::onlyTrashed()->find($model->user_id)->first_name;
+                                $last = User::onlyTrashed()->find($model->user_id)->last_name;
```
Comment (mverma16) [inline app/Http/Controllers/Order/InvoiceController.php:146]: use negating condition for improving readability

**Review by mverma16 — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/Order/InvoiceController.php (hunk 1 above)
Comment (ashu555) [reply app/Http/Controllers/Order/InvoiceController.php:146]: fixed

**Comment by mverma16** [comment]
Do not see any change since I last reviewed, have you pushed the changes?

**Review by mverma16 — APPROVED** [review]

**Review by SarfrazS — APPROVED** [review]
when we suspended users that time users' invoices and orders are not deleted. Once we deleted permanently than the user's invoices and orders all are deleted.

Working fine.

**Pull request closed without merging**